src/base/net/geoipmanager.cpp

```diff
@@ -392,10 +401,15 @@ QString GeoIPManager::CountryName(const QString &countryISOCode)
 
 void GeoIPManager::configure()
 {
-    const bool enabled = Preferences::instance()->resolvePeerCountries();
-    if (m_enabled != enabled) {
+    const Preferences *const pref = Preferences::instance();
+    const bool enabled = pref->resolvePeerCountries();
+    QString license = pref->geoIPLicenseKey();
+    QString dbCompleteUrl = license.isEmpty() ? QString {} : QString {DATABASE_URL + license};
+    if ((m_enabled != enabled) || (m_dbCompleteUrl != dbCompleteUrl)) {
         m_enabled = enabled;
-        if (m_enabled && !m_geoIPDatabase) {
+        if (m_enabled
+            && (!m_geoIPDatabase || (m_dbCompleteUrl != dbCompleteUrl))) {
```

Comment:
These conditions seem to be redundant.
Due to conditions in "if" above it was either disabled before (so m_geoIPDatabase was nullptr) or it had different database url.

Reply:
>either disabled before (**so m_geoIPDatabase was nullptr**)

I didn't get into the specifics because the deleted check included `!m_geoIPDatabase` already. Of the top of my head I can't find a flow where it results that it was disabled but with non-nullptr pointer (m_geoIPDatabase). Maybe it was put there as an extra precaution?

Should I try to remove it? (and from the else if below it?)

Reply:
>Should I try to remove it? (and from the else if below it?)

It can be written as:
```c++
if (m_enabled) {
}
else {
}
```
without any other conditions.
>Maybe it was put there as an extra precaution?

Maybe...
Anyway we shouldn't handle it as regular case (unless we really want to keep database loaded after feature is disabled).